QuizApp/Data Layer/Models/Quiz.swift

```diff
@@ -17,4 +20,12 @@ struct Quiz: Codable {
         case imageUrl = "image"
         case questions
     }
+    
+    mutating func getImage() {
+        guard imageData == nil else { return }
+        
+        if let url = URL.init(string: imageUrl) {
+            imageData = try? Data(contentsOf: url)
+        }
+    }
```

Comment:
Try to avoid `mutating` functions on models. Swift has a very good way of creating and destroying structs so it's a cheap operation and it helps with having a thread-safe code since no two threads will use the same object.
Also, loading data is not something a model object should be doing. The main layer class (in this case datasource) should fetch everything and create a model.